src/app/home/user/components/detail/detail.component.ts

```diff
@@ -0,0 +1,78 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup } from '@angular/forms';
+import { MatDialog, MatDialogConfig } from '@angular/material/dialog';
+import { TaskService } from '../../services/task.service';
+import { UserActivityComponent } from '../user-activity/user-activity.component';
+
+@Component({
+  selector: 'app-detail',
+  templateUrl: './detail.component.html',
+  styleUrls: ['./detail.component.css'],
+})
+export class DetailComponent implements OnInit {
+
+  public userDetails: FormGroup;
+
+  constructor(
+    private taskService: TaskService,
+    private fb: FormBuilder,
+    private dialog: MatDialog
+  ) {
+    this.userDetails = this.fb.group({
+      heightUnit: new FormControl('cm'),
+    });
+  }
+
+  username: string = '';
+  userData: object = {};
+  keys: any[] = [];
+  values: any[] = [];
+  toggleButton = true;
+
+  ngOnInit(): void {
+    const user = localStorage.getItem('username');
```

Comment:
Move the logic to get any value from local storage to service.
Preferably make a common service in /app level to handle basic stuff like fetch local storage item